src/deblock.rs

```diff
@@ -53,180 +53,229 @@ fn deblock_adjusted_level(deblock: &DeblockState, block: &Block, pli: usize, ver
-            p3: i32, p2: i32, p1: i32, p0: i32, q0: i32, q1: i32, q2: i32, q3: i32) {
-    data[pitch]   = (p3*3 + p2*2 + p1 + p0 + q0 + (1<<2) >> 3) as u16;
-    data[pitch*2] = (p3*2 + p2 + p1*2 + p0 + q0 + q1 + (1<<2) >> 3) as u16;
-    data[pitch*3] = (p3 + p2 + p1 + p0*2 + q0 + q1 + q2 + (1<<2) >> 3) as u16;
-    data[pitch*4] = (p2 + p1 + p0 + q0*2 + q1 + q2 + q3 + (1<<2) >> 3) as u16;
-    data[pitch*5] = (p1 + p0 + q0 + q1*2 + q2 + q3*2 + (1<<2) >> 3) as u16;
-    data[pitch*6] = (p0 + q0 + q1 + q2*2 + q3*3 + (1<<2) >> 3) as u16;
+// six taps, 4 outputs
+fn filter_wide6(p2: i32, p1: i32, p0: i32, q0: i32, q1: i32, q2: i32) -> [i32; 4] {
+    [p2*3 + p1*2 + p0*2 + q0   + (1<<2) >> 3,
+     p2   + p1*2 + p0*2 + q0*2 + q1   + (1<<2) >> 3,
+            p1   + p0*2 + q0*2 + q1*2 + q2   + (1<<2) >> 3,
+                   p0   + q0*2 + q1*2 + q2*3 + (1<<2) >> 3]
 }
 
-fn filter_wide14(data: &mut [u16], pitch: usize,
-             p6: i32, p5: i32, p4: i32, p3: i32, p2: i32, p1: i32, p0: i32,
-             q0: i32, q1: i32, q2: i32, q3: i32, q4: i32, q5: i32, q6: i32) {
-    data[pitch]   = (p6*7 + p5*2 + p4*2 + p3 + p2 + p1 + p0 + q0 + (1<<3) >> 4) as u16;
-    data[pitch*2] = (p6*5 + p5*2 + p4*2 + p3*2 + p2 + p1 + p0 + q0 + q1 + (1<<3) >> 4) as u16;
-    data[pitch*3] = (p6*4 + p5 + p4*2 + p3*2 + p2*2 + p1 + p0 + q0 + q1 + q2 + (1<<3) >> 4) as u16;
-    data[pitch*4] = (p6*3 + p5 + p4 + p3*2 + p2*2 + p1*2 + p0 + q0 + q1 + q2 + q3 + (1<<3) >> 4) as u16;
-    data[pitch*5] = (p6*2 + p5 + p4 + p3 + p2*2 + p1*2 + p0*2 + q0 + q1 + q2 + q3 + q4 + (1<<3) >> 4) as u16;
-    data[pitch*6] = (p6 + p5 + p4 + p3 + p2 + p1*2 + p0*2 + q0*2 + q1 + q2 + q3 + q4 + q5 + (1<<3) >> 4) as u16;
-    data[pitch*7] = (p5 + p4 + p3 + p2 + p1 + p0*2 + q0*2 + q1*2 + q2 + q3 + q4 + q5 + q6 + (1<<3) >> 4) as u16;
-    data[pitch*8] = (p4 + p3 + p2 + p1 + p0 + q0*2 + q1*2 + q2*2 + q3 + q4 + q5 + q6*2 + (1<<3) >> 4) as u16;
-    data[pitch*9] = (p3 + p2 + p1 + p0 + q0 + q1*2 + q2*2 + q3*2 + q4 + q5 + q6*3 + (1<<3) >> 4) as u16;
-    data[pitch*10] = (p2 + p1 + p0 + q0 + q1 + q2*2 + q3*2 + q4*2 + q5 + q6*4 + (1<<3) >> 4) as u16;
-    data[pitch*11] = (p1 + p0 + q0 + q1 + q2 + q3*2 + q4*2 + q5*2 + q6*5 + (1<<3) >> 4) as u16;
-    data[pitch*12] = (p0 + q0 + q1 + q2 + q3 + q4*2 + q5*2 + q6*7 + (1<<3) >> 4) as u16;
+// eight taps, 6 outputs
+fn filter_wide8(p3: i32, p2: i32, p1: i32, p0: i32, q0: i32, q1: i32, q2: i32, q3: i32) -> [i32; 6] {
+    [p3*3 + p2*2 + p1   + p0   + q0   + (1<<2) >> 3,
+     p3*2 + p2   + p1*2 + p0   + q0   + q1   + (1<<2) >> 3,
+     p3   + p2   + p1   + p0*2 + q0   + q1   + q2   +(1<<2) >> 3,
+            p2   + p1   + p0   + q0*2 + q1   + q2   + q3   + (1<<2) >> 3,
+                   p1   + p0   + q0   + q1*2 + q2   + q3*2 + (1<<2) >> 3,
+                          p0   + q0   + q1   + q2*2 + q3*3 + (1<<2) >> 3]
+}
+
+// fourteen taps, 12 outputs
+fn filter_wide14(p6: i32, p5: i32, p4: i32, p3: i32, p2: i32, p1: i32, p0: i32,
```

Comment:
Make sure to add `#[cfg_attr(rustfmt, rustfmt_skip)]` to preserve the nicer formatting.